static/app/views/issueDetails/groupTagValues.tsx

```diff
@@ -43,55 +37,109 @@ type RouteParams = {
 type Props = {
   baseUrl: string;
   group: Group;
-  organization: Organization;
   environments?: string[];
   project?: Project;
-} & RouteComponentProps<RouteParams, {}>;
-
-type State = {
-  tag: Tag | null;
-  tagValueList: TagValue[] | null;
-  tagValueListPageLinks: string;
 };
 
 const DEFAULT_SORT = 'count';
 
-class GroupTagValues extends DeprecatedAsyncComponent<
-  Props & DeprecatedAsyncComponent['props'] & WithRouterProps,
-  State & DeprecatedAsyncComponent['state']
-> {
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {environments: environment, organization} = this.props;
-    const {groupId, tagKey} = this.props.params;
-    return [
-      ['tag', `/organizations/${organization.slug}/issues/${groupId}/tags/${tagKey}/`],
-      [
-        'tagValueList',
-        `/organizations/${organization.slug}/issues/${groupId}/tags/${tagKey}/values/`,
-        {query: {environment, sort: this.getSort()}},
-      ],
-    ];
-  }
-
-  getSort(): string {
-    return this.props.location.query.sort || DEFAULT_SORT;
-  }
+function useTagQueries({
+  group,
+  tagKey,
+  environments,
+  sort,
+}: {
+  group: Group;
+  sort: string | string[];
+  environments?: string[];
+  tagKey?: string;
+}) {
+  const organization = useOrganization();
+
+  const {
+    data: tagValueList,
+    isLoading: tagValueListIsLoading,
+    isError: tagValueListIsError,
+    getResponseHeader,
+  } = useApiQuery<TagValue[]>(
+    [
+      `/organizations/${organization.slug}/issues/${group.id}/tags/${tagKey}/values/`,
+      {query: {environment: environments, sort}},
+    ],
+    {staleTime: 0, retry: false}
+  );
+  const {data: tag, isError: tagIsError} = useApiQuery<Tag>(
```

Comment:
Should this be composed of two different hooks (useGroupTag and useGroupTags)? I don't know if this API is used in other places so this might be a moot point, but it would make reuse easier. It would probably make sense for us to think of a convention for data fetching hooks so we don't end up duplicating them and make them easier to discover. Would love your and @scttcper's thoughts on that since I see you two have been converting plenty of these :)

Reply:
we do seem to have a fetchTagValues that this overlaps with https://github.com/getsentry/sentry/blob/master/static/app/actionCreators/tags.tsx#L87 and could be shared later

Reply:
Good callout - didn't realize how many other components were making this same request. Definitely better to make these reusable when we can